File: 2020/15-2.js
fs = require("fs");

var v, m = {};
for(var i = 0; i < 24e6; i++) m[i] = -1;

var s = fs.readFileSync("15.txt").toString().trim()
	.split(",").map((t, i) => (m[t] = i, v=1*t));
console.log(s)

for(var i = s.length; i < 3e7; i++){
	t = m[v] == -1 || m[v] === undefined ? 0 : i - m[v] - 1;
	m[v] = i - 1;
	v = t;
	// if(!(i%3e6))console.log(100*i/30000000|0)
}

console.log(v);

File: 2020/15-2.pl
use strict;
use warnings;
use v5.15.1;

my ($i, $v, %m) = (0, 0, ());
open(FILE, '<', "15.txt") or die $!;
my @l = map {$m{(int)} = $i++; $v = int} split ",", <FILE>;
close(FILE);

for ($i = @l; $i < 3e7; $i++) {
	my $t = defined($m{$v}) ? $i - $m{$v} - 1 : 0;
	$m{$v} = $i - 1;
	$v = $t;
	$i % 3e6 || $mu->record((100*$i/3e7)."%");
}

say $v;
